From 1593c91e1daa4ea5d4b532f877ad5560a202f4bd Mon Sep 17 00:00:00 2001
From: Avery Pennarun <apenwarr@gmail.com>
Date: Sun, 25 Mar 2012 19:11:34 -0400
Subject: [PATCH] Fix dependency on tc/emp_ematch.yacc.h.

emp_ematch.lex.c includes emp_ematch.yacc.h, which is generated by running
bison.  So we have to make emp_ematch.lex.o depend on it.

While we're here, fix up 'make clean' to actually remove all the
yacc-generated files.
---
 tc/Makefile |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/tc/Makefile b/tc/Makefile
index c9d7479..3012a99 100644
--- a/tc/Makefile
+++ b/tc/Makefile
@@ -117,7 +117,7 @@ install: all
 	fi
 
 clean:
-	rm -f $(TCOBJ) $(TCLIB) libtc.a tc *.so emp_ematch.yacc.h; \
+	rm -f $(TCOBJ) $(TCLIB) libtc.a tc *.so emp_ematch.yacc.[ch]; \
 	rm -f emp_ematch.yacc.output
 
 q_atm.so: q_atm.c
@@ -129,12 +129,14 @@ m_xt.so: m_xt.c
 m_xt_old.so: m_xt_old.c
 	$(CC) $(CFLAGS) $(LDFLAGS) -shared -fpic -o m_xt_old.so m_xt_old.c -lxtables
 
-%.yacc.c: %.y
-	$(YACC) $(YACCFLAGS) -o $@ $<
+%.yacc.c %.yacc.h: %.y
+	$(YACC) $(YACCFLAGS) -o $*.yacc.c $<
 
 %.lex.c: %.l
 	$(LEX) $(LEXFLAGS) -o$@ $<
 
+emp_ematch.lex.c: emp_ematch.yacc.h
+
 ifneq ($(SHARED_LIBS),y)
 
 tc: static-syms.o
-- 
1.7.7.3

